internal/request/request.go

```diff
@@ -111,17 +111,20 @@ func startDispatcher(ctx context.Context, config *common.CatalogConfig, wc tower
 }
 
 type pageWriterFactory interface {
-	makePageWriter(ctx context.Context, format string, uploadURL string, task catalogtask.CatalogTask) (PageWriter, error)
+	makePageWriter(ctx context.Context, format string, uploadURL string, task catalogtask.CatalogTask, taskURL string) (PageWriter, error)
```

Comment:
Looks like we should take a map instead of taskURL. The factory does not need to convert it to a map.